tests/integration/dev.bats

```diff
@@ -33,3 +33,46 @@ function teardown() {
 	[ "$status" -eq 0 ]
 	[[ "${lines[0]}" =~ "crw-rw-rw".+"1".+"root".+"root".+"5,".+"2".+"/dev/ptmx" ]]
 }
+
+@test "runc run [device cgroup allow]" {
+	[[ "$ROOTLESS" -ne 0 ]] && requires root
+
+	update_config ' .linux.resources.devices = [{"allow": false, "access": "rwm"},{"allow": true, "type": "c", "major": 1, "minor": 11, "access": "rw"}]
+			| .linux.devices = [{"path": "/dev/kmsg", "type": "c", "major": 1, "minor": 11}]
+			| .process.args |= ["sh"]
+			| .hostname = "myhostname"'
+
+	runc run -d --console-socket "$CONSOLE_SOCKET" test_allow
+	[ "$status" -eq 0 ]
+
+	# test write
+	runc exec test_allow sh -c 'hostname | tee /dev/kmsg'
+	[ "$status" -eq 0 ]
+	[[ "${lines[0]}" == *'myhostname'* ]]
+
+	# test read
+	runc exec test_allow sh -c 'head -n 1 /dev/kmsg'
+	[ "$status" -eq 0 ]
+}
+
+@test "runc run [device cgroup deny]" {
+	[[ "$ROOTLESS" -ne 0 ]] && requires root
+
+	update_config ' .linux.resources.devices = [{"allow": false, "access": "rwm"}]
+			| .linux.devices = [{"path": "/dev/kmsg", "type": "c", "major": 1, "minor": 11}]
```

Comment:
Can we have a test for block devices as well? (I think the previous revision had block device tests?)

Reply:
sure, let me add that back. 

When working on this, I saw some errors and I got questions about it
1. would you think that is a common scenario to add devices in a different path that the one in the host? for example host: `/dev/null` and in the container: `/dev/mynull`. Doing this in a container with user namespace, I would get this error `rootfs_linux.go:72: creating device nodes caused: no such file or directory`. If I am not wrong it happens because they are bind mounted usign the `device.Path` instead of created with mknod.

2. and also, is a common scenario to add specific deny rules besides deny all? when RUNC_USE_SYSTEMD is set to yes and I add something like `{"allow": true, "access": "rwm"}, {"allow": false, "access": "rw", "type": "c", "major": 1, "minor": 11}` I got `systemd doesn't support blacklist device rules -- applying temporary deny-all rule` and then I go this error `container init caused: rootfs_linux.go:123: reopening /dev/null inside container caused: Failed to open /dev/null - open /dev/null: operation not permitted`

thanks

Reply:
You can just skip the failing tests (with code comments) for userns and systemd

Reply:
Added one more integration test that tests the read and mknod operations for a block device